### ISSUE TITLE
getattr not falling back to default with custom __getattr__

### ISSUE THREAD

**Issue by dnr** [micropython/micropython#6089]
Here's an example:

```
class tester:
  def __init__(self):
    self.data = {'a': 5}

  def __getattr__(self, n):
    if n in self.data:
      return self.data[n]
    else:
      raise AttributeError(n)

t = tester()

print(getattr(t, 'a', 'default'))
print(getattr(t, 'b', 'default'))
```

In CPython, that prints `5`, `default`. In MicroPython (1.12), it prints `5`, but then the `AttributeError` gets raised to the top level and `getattr` doesn't return `default`.

**Comment by dpgeorge** [comment]
Thanks for the report, indeed it is an inconsistency with CPython.

See #6090 for a proposed fix.

**Comment by dpgeorge** [comment]
Fixed by 203b10703e5df20939b88a2ce29bb961cce7e4b6
